core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -368,6 +369,7 @@ export class ExecuteWorkflowService {
       case ExecutionState.Resuming:
       case ExecutionState.Running:
       case ExecutionState.Initializing:
+        this.workflowActionService.toggleLockListen(false);
```

Comment:
why do we turn lock listen off during execution?

Reply:
Because if it is still listening to the lock change, during execution if another session acquires the lock, workflow action change will be enabled. `lockListen` is only an internal variable for `workflowActionService` to turn on/off actions.

Reply:
During execution, I wonder if anyone should be able to obtain write lock?

Reply:
I think so because during execution maybe another user may wants to pause the workflow.

Reply:
This might lead to a separation of execution mode and editing mode. I feel during execution mode, no edit should be done by anyone. But one with the write lock may control the execution (like pause). If paused, it returns to edit mode. 

Not sure if the above is sufficient. This might need further discussion with the group.

Reply:
So for the short term, I suggest that the one with no write lock cannot do anything to the workflow, besides viewing. He may request for the write lock to "pause" it. Is this the current behavior? I will also test myself later.

Reply:
Yes. That is the current behaviour